xcontour/utils.py

```diff
@@ -124,35 +126,87 @@ def add_latlon_metrics(dset, dims=None, boundary=None, Rearth=Rearth):
         # dlonG = grid.diff(lonG, 'X', boundary_discontinuity=360)
         dlonC = grid.diff(lonC, 'X')   
         dlonG = grid.diff(lonG, 'X')
-        dlonC[0 ] = dlonC[0 ] + 360 # mini-dong: adjust initial point
-        dlonG[-1] = dlonG[-1] + 360 # mini-dong: adjust initial point
+        for i in [0,-1]:  # mini-dong: adjust periodic terminal point
```

Comment:
We still need to use loops here, because the index of -1 may be negative. I will change it like:
```python
for i in [0,-1]:  # mini-dong: adjust periodic terminal point
    if dlonC[i ] < 0:
        dlonC[i ] += 360
    elif dlonC[i ] > 360:
        dlonC[i ] -= 360
    if dlonG[i ] < 0:
        dlonG[i ] += 360
    elif dlonG[i ] > 360:
        dlonG[i ] -= 360       
```